consensus/consensus.md

```diff
@@ -413,24 +413,40 @@ We can estimate the network capacity used to mine a given block `i` as
 
 ```
 EstimatedCapacity[i] = Difficulty[i] / MiningTime[i]
-MiningTime[i]        = Time[i + 1] - Time[i]
+MiningTime[i]        = Time[i] - Time[i - 1]
 ```
 
 The estimated capacity across all `N` blocks is then the weighted (by time)
-average of the estimated capacities for each block.
+average of the estimated capacities for each block. Note, since `MiningTime[i]`
```

Comment:
I see patch pushed:

> Note, since `MiningTime[i]`
> require `Time[i - 1]` and we do not use the genesis block timestamp in target
> adjustment we cannot start using the target estimation until block `N + 2 (= 19)`.

Reply:
As genesis block has height 0, the first 17 blocks to consider are from 2 to 18 (hence prev timestamps 1 to 17). So first block that shall have target not necessarily equal to predecessor is 19 - correct.

Reply:
Current code in verification expects that [block 18 has target copied](https://github.com/aeternity/epoch/blob/0eb614be03ce0cccbbdf7c58420bbdda7b739249/apps/aecore/src/aec_chain_state.erl#L366) that seems correct.

Reply:
Current code in generation [if top hash height 18](https://github.com/aeternity/epoch/blob/0eb614be03ce0cccbbdf7c58420bbdda7b739249/apps/aecore/src/aec_block_key_candidate.erl#L55) (hence candidate will have height 19) it triggers target adjustment.

Reply:
So code looks correct (19 first block with adjusted target) in both verification and generation.